src/C++/SSLSocketAcceptor.cpp

```diff
@@ -356,6 +357,11 @@ void SSLSocketAcceptor::onWrite( SocketServer& server, socket_handle s )
   SocketConnections::iterator i = m_connections.find( s );
   if ( i == m_connections.end() ) return ;
   SSLSocketConnection* pSocketConnection = i->second;
+
+  if (pSocketConnection->didReadFromSocketRequestToWrite()) {
+     pSocketConnection->read(*this, server);
+  }
+
```

Comment:
OpenSSL uses a state-machine which maintains an input buffer and an output buffer. When read is called (from onData) it could be than SSL_read fails with error: SSL_ERROR_WANT_WRITE. This is because SSL may want to renegotiate keys. 
In case this happens, we set a flag (to indicate that readFromSocket method requested to write) and exit that method. 
In method (onWrite) we check if the flag is set, then we call read again which will resume the OpenSSL state-machine.